### PR TITLE
depends: Remove ccache

### PR DESCRIPTION
**Pull request by fanquake**
After discussion with @theuni, we can possibly just remove ccache from depends entirely.

Related to #12606

### CODE REVIEW

**Comment by laanwj** [comment]
Concept ACK, let's see what Travis says

**Comment by practicalswift** [comment]
utACK ec7602b39eea0a7f8bc59eca3ac530e96e1e90e6

**Comment by MarcoFalke** [comment]
Looks like configure picked up the ccache: `checking if ccache should be used... yes`

utACK ec7602b

**Comment by jonasschnelli** [comment]
Lets see the gitian builder says: https://bitcoin.jonasschnelli.ch/build/524

**Comment by MarcoFalke** [comment]
@jonasschnelli 's gitian build seems fine

**Comment by laanwj** [comment]
What we still need to check is whether travis caches the ccache result, even though `~/.ccache` is no longer in the list. I'm not sure how, is it possible to inspect the travis cache?

File: .travis.yml [hunk 1]
```diff
@@ -3,10 +3,10 @@ dist: trusty
 os: linux
 language: minimal
 cache:
+  ccache: true
   directories:
   - depends/built
   - depends/sdk-sources
-  - $HOME/.ccache
```
Comment (laanwj) [inline .travis.yml]: My guess is that this needs to be re-added

**Comment by randolf** [comment]
Just in case it's helpful, I successfully complied this on NetBSD 7.0 (64-bit).  These are the commands I used:

``git clone https://www.github.com/fanquake/bitcoin/``
``cd bitcoin``
``git checkout ccache-removal``
``./autogen.sh``
``./configure CPPFLAGS="-I/usr/pkg/include" LDFLAGS="-L/usr/pkg/lib" BOOST_CPPFLAGS="-I/usr/pkg/include" BOOST_LDFLAGS="-L/usr/pkg/lib"``
``gmake``

If you'd like me to try building with different parameters, or check for the presence of certain files/directories, please let me know.

**Comment by laanwj** [comment]
@randolf Thanks for testing, however this does not affect normal autotools builds, if you don't use the depends system. And I don't think the depends system works on NetBSD (though never tried).

**Comment by randolf** [comment]
@laanwj I don't know enough about the "depends system" to provide helpful perspective there, but I'd be happy to try if there are some instructions somewhere that I can follow.

**Review by theuni — APPROVED** [review]
ACK cc879675e16c310f3ce3f2387b7aa966e64e4352

**Comment by MarcoFalke** [comment]
re-utACK cc879675e16c310f3ce3f2387b7aa966e64e4352

Caching works on arm and osx cross-builds at least

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/12607/commits/cc879675e16c310f3ce3f2387b7aa966e64e4352 - will merge this tomorrow after the meeting to minimize the amount of interference with ongoing travis testing

**Comment by MarcoFalke** [comment]
@laanwj I assume the meeting is over?

**Comment by ryanofsky** [comment]
The travis builds on all my PR's seem to be failing with errors like:

```
$ if [ -z "$NO_DEPENDS" ]; then depends/$HOST/native/bin/ccache --max-size=$CCACHE_SIZE; fi
/home/travis/.travis/job_stages: line 57: depends/x86_64-unknown-linux-gnu/native/bin/ccache: No such file or directory
```

An example is https://travis-ci.org/bitcoin/bitcoin/jobs/350509177#L1055

**Comment by MarcoFalke** [comment]
@ryanofsky This is a travis bug. The command you mention above was removed from the travis yaml, it shouldn't be executed at all.

You can either ignore the failure or rebase your pull request on master. (Just changing the commit id via `EDITOR=true git commit --ammend` might be enough, but I haven't checked)

**Comment by MarcoFalke** [comment]
Just checked that rebase is not needed. You can just change the commit id

**Pull request merged**